include/assertion.h

```diff
@@ -0,0 +1,160 @@
+    fido_assert_auth_data_t auth_data;
+    uint8_t                 signature[ASSERTION_SIGNATURE_LENGTH];
+    uint8_t                 large_blob_key[LARGEBLOB_KEY_SIZE];
+    bool                    has_large_blob_key;
+} fido_assert_reply_t;
+
+
+typedef struct fido_assert {
+    fido_assert_blob_t          rp_id;                                  // relying party id
+    uint8_t                     cdh[ASSERTION_CLIENT_DATA_HASH_LEN];    // client data hash
+    fido_assert_opt_t           opt;                                    // user presence & user verification
+    fido_assert_ext_t           ext;                                    // enabled extensions
+    fido_assert_reply_t         reply;                                  // The parsed reply. Only one credential is supported!
+} fido_assert_t;
+
+/**
+ * @brief Reset an assertion request to a known state.
+ * 
+ * @param assert A pointer to the assertion data structure to reset.
+ */
+void fido_assert_reset(fido_assert_t *assert);
+
+/**
+ * @brief Get assertion from device.
+ *
+ * Note that only one assertion statement is supported (numberOfCredentials > 1 is ignored).
+ *
+ * @param dev The device to read from.
+ * @param assert Options for the assertion.
+ * @return success or failure
+ */
+int fido_dev_get_assert(fido_dev_t *dev, fido_assert_t *assert);
+
+/**
+ * @brief Set the relying party ID for an assertion.
+ *
+ * **Warning:** Do not change the string data until having called `fido_dev_get_assert`.
+ *
+ * @param assert A pointer to an assertion request to set the relying party on.
+ * @param id The relying party ID as string (NULL-terminated).
+ */
+void fido_assert_set_rp(fido_assert_t *assert, const char* id);
+
+/**
+ * @brief Set the client data hash for an assertion.
+ *
+ * Note: This should be non-predictable.
+ *
+ * @param assert A pointer to an assertion request to set the client data hash on.
+ * @param hash The SHA256 hash of the client data.
+ */
+void fido_assert_set_client_data_hash(fido_assert_t *assert, const uint8_t hash[ASSERTION_CLIENT_DATA_HASH_LEN]);
+
+/**
+ * @brief Set the options field for an assertion.
+ *
+ * @param assert A pointer to an assertion request to set the options on.
+ * @param options The options, see FIDO_ASSERT_OPTION_*.
+ */
+void fido_assert_set_options(fido_assert_t *assert, const fido_assert_opt_t options);
+
+/**
+ * @brief Set the extensions field for an assertion.
+ *
+ * @param assert A pointer to an assertion request to set the extensions on.
+ * @param extensions The extensions, see FIDO_ASSERT_EXTENSION_*.
+ */
+void fido_assert_set_extensions(fido_assert_t *assert, const fido_assert_ext_t extensions);
+
+/**
+ * @brief Verify an assertion.
+ * 
+ * @param assert A pointer to an assertion request/reply struct.
+ * @param cose_alg A COSE algorithm identifier.
+ * @param pk The key to verify the signature.
+ * @return int 
+ */
+int fido_assert_verify(const fido_assert_t *assert, int cose_alg, const void *pk);
```

Comment:
I think `pk` will never be not `uint8_t*`. Is there any special reason for the `void*`?

Reply:
Nope, changed to `uint8_t*`.